src/core.js

```diff
@@ -917,3 +917,5 @@ const processPreload = link => {
     fetchCache[link.href] = fetchModule(link.href, getFetchOpts(link));
   });
 };
+
+if (hotReloadEnabled) initHotReload(topLevelLoad, importShim);
```

Comment:
Previously the hooks would be initialized before we got here. Would be worth verifying there is no sync use of hooks here before that. Otherwise this should be at the top of the file with the topLevelLoad as a hoisted function and importShim maybe defined sooner to support that.